### PR TITLE
TypeScript Integration

### PR DESCRIPTION
**Pull request by asutula**
Feeling really good about using TS. I migrated our `UIRedux` over to TS and love the way it works. Also doesn't seem to be a problem to mix TS and regular JS code, so we can write new code in TS and migrate old code as needed.

Also organized a Jest testing configuration a bit, let's get the test coverage in here way up!

### CODE REVIEW

**Comment by andrewxhill** [comment]
officially maintaining more languages than team members! haha, love the TS though

**Comment by asutula** [comment]
Haha yea true, but this will be a good change :)

File: App/Redux/__tests__/UIRedux.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import actions, { reducer } from '../UIRedux'
+
+const initialState = reducer(undefined, {} as any)
+
+describe('ui stories', () => {
```
Comment (asutula) [inline App/Redux/__tests__/UIRedux.ts]: damn writing tests in TS with Jest is a total pleasure!

Comment (sanderpick) [reply App/Redux/__tests__/UIRedux.ts]: awesome

**Comment by sanderpick** [comment]
seeing a failing test though

**Comment by asutula** [comment]
Yea failing test looks related to the version of node on the macOS build server image from CircleCI. Trying a fix now...

**Comment by asutula** [comment]
Cool, CI passing now @sanderpick

**Pull request merged**